grammar.js

```diff
@@ -1640,16 +1640,101 @@ module.exports = grammar({
     void_keyword: $ => 'void',
 
     preprocessor_call: $ => seq(
-      $.preprocessor_directive,
-      repeat(choice(
-        $.identifier,
-        $._literal,
-        token(prec(-1, /[^\s]+/))
-      )),
+      $._preproc_directive_start,
+      choice(
+          $.nullable_directive,
+          $.define_directive,
+          $.undef_directive,
+          $.if_directive,
+          $.else_directive,
+          $.elif_directive,
+          $.endif_directive,
+          $.region_directive,
+          $.endregion_directive,
+          $.error_directive,
+          $.warning_directive,
+          $.line_directive,
+          $.pragma_directive
+      ),
       $._preproc_directive_end
     ),
 
-    preprocessor_directive: $ => /#[ \t]*[a-z]\w*/,
+    _preproc_directive_start: $ => /#[ \t]*/,
```

Comment:
I don't think we should support ScriptCS - it has a bunch of different grammar rules and with the top-level statements I think main CS will replace it now.

We should support that `#!` tho, I'll add there. Where did you spot that (in case I missed others)?